src/Facility.CodeGen.CSharp/CSharpGenerator.cs

```diff
@@ -252,6 +252,21 @@ private CodeGenFile GenerateEnum(ServiceEnumInfo enumInfo, Context context)
 
 						code.WriteLine();
 						code.WriteLine("readonly string m_value;");
+
+						code.WriteLine();
+						code.WriteLine("public static class Strings");
```

Comment:
I'm not sure how if an [obsolete] on the nest parent of the class is sufficient. Maybe this class needs one too?

Reply:
Though, I suppose marking individual constants obsolete makes sense, because the constant should be obsolete if its enum variant is and a single (or subset) of the variants may be marked as such.